### PR TITLE
Follow the CommonWith pattern in compilation options

### LINKED ISSUE #5953: Determinism option needs to follow the CommonWith pattern
The Determinism option needs to follow the CommonWith pattern:

- CompilationOptions has an abstract CommonWithDeterminism member 
- CompilationOptions has a non-virtual WithDeterminism member 
- CSharp and VB specific CompilationOptions overrides CommonWithDeterminism and shadows WithDeteriminism 

See CommonWithDocumentationMode, or CommonWithKind as an example here.  

My bad for not spotting this on the initial PR.

### PR DESCRIPTION
**Pull request by TyOverby**
Closes #5953

### CODE REVIEW

**Comment by TyOverby** [comment]
Please review: @jaredpar @dotnet/roslyn-compiler

File: src/Compilers/Core/Portable/Compilation/CompilationOptions.cs [hunk 1]
```diff
@@ -361,6 +361,14 @@ public CompilationOptions WithReportSuppressedDiagnostics(bool value)
         }
 
         /// <summary>
+        /// Creates a new options instance with the deterministic property set accordingly.
+        /// </summary>
+        public CompilationOptions WithDeterministic(bool deterministic)
```
Comment (agocke) [inline src/Compilers/Core/Portable/Compilation/CompilationOptions.cs]: @jaredpar Shouldn't this be throwing up some public api change?

Comment (jaredpar) [inline src/Compilers/Core/Portable/Compilation/CompilationOptions.cs]: @TyOverby this should have triggered an API change.  Did it not? 

Also I would have expected the CSharpCompilationOptions to require "new" on the shadow of this method.

Comment (TyOverby) [inline src/Compilers/Core/Portable/Compilation/CompilationOptions.cs]: It probably should?  No errors on my side though.

Comment (TyOverby) [inline src/Compilers/Core/Portable/Compilation/CompilationOptions.cs]: @jaredpar No API change error (or warning in VS).  

You were right about the "new" shadow.  Vs did catch that one.  I think I ran my build incorrectly the first time through, but I caught it and changed it when I went back and looked.

**Pull request merged**